src/Framework/Validation/Rules/Max.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Give\Framework\Validation\Rules;
+
+use Closure;
+use Give\Framework\Exceptions\Primitives\InvalidArgumentException;
+use Give\Framework\Validation\Contracts\ValidatesOnFrontEnd;
+use Give\Framework\Validation\Contracts\ValidationRule;
+use Give\Framework\Validation\Exceptions\ValidationException;
+
+/**
+ * @unreleased
+ */
+class Max implements ValidationRule, ValidatesOnFrontEnd
+{
+    /**
+     * @var int
+     */
+    private $size;
+
+    /**
+     * @inheritDoc
+     *
+     * @unreleased
+     */
+    public static function id(): string
+    {
+        return 'max';
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @unreleased
+     */
+    public static function fromString(string $options = null): ValidationRule
+    {
+        if (!is_numeric($options)) {
+            throw new InvalidArgumentException('Max validation rule requires a numeric value');
+        }
+
+        return new self((int)$options);
+    }
+
+    /**
+     * @unreleased
+     */
+    public function __construct(int $size)
+    {
+        if ($size <= 0) {
+            throw new InvalidArgumentException('Max validation rule requires a non-negative value');
+        }
+
+        $this->size = $size;
+    }
```

Comment:
I noticed _some_ rules are using a `__construct` and it appears because they store some values internally.  Should we add some kind of contract to enforce the construct for these types of rules?

Reply:
I thought about this and didn't see any benefit in constraining the constructors. They're never statically instantiated. I like the flexibility until we otherwise have a solid reason to reduce it.